src/lib.rs

```diff
@@ -232,40 +205,34 @@
 //! ```
 
 #![no_std]
-#![warn(future_incompatible)]
+#![warn(future_incompatible, rust_2018_idioms)]
 #![deny(missing_docs)] // refuse to compile if documentation is missing
 #![cfg_attr(not(test), forbid(unsafe_code))]
 
+#[cfg(any(feature = "batch", feature = "batch_deterministic"))]
+extern crate alloc;
+
 #[cfg(any(feature = "std", test))]
 #[macro_use]
 extern crate std;
 
-pub extern crate ed25519;
-
-#[cfg(all(feature = "alloc", not(feature = "std")))]
-extern crate alloc;
-extern crate curve25519_dalek;
-#[cfg(all(any(feature = "batch", feature = "batch_deterministic"), any(feature = "std", feature = "alloc")))]
-extern crate merlin;
-#[cfg(any(feature = "batch", feature = "std", feature = "alloc", test))]
-extern crate rand;
 #[cfg(feature = "serde")]
 extern crate serde_crate as serde;
```

Comment:
Can't get rid of this yet, unfortunately, but we can probably get rid of it by switching to namespaced features